include/mbgl/style/expression/dsl.hpp

```diff
@@ -31,14 +31,21 @@ std::unique_ptr<Expression> literal(Value value);
 std::unique_ptr<Expression> literal(std::initializer_list<double> value);
 std::unique_ptr<Expression> literal(std::initializer_list<const char *> value);
 
-std::unique_ptr<Expression> assertion(type::Type, std::unique_ptr<Expression>);
-std::unique_ptr<Expression> number(std::unique_ptr<Expression>);
-std::unique_ptr<Expression> string(std::unique_ptr<Expression>);
-std::unique_ptr<Expression> boolean(std::unique_ptr<Expression>);
-
-std::unique_ptr<Expression> toColor(std::unique_ptr<Expression>);
-std::unique_ptr<Expression> toString(std::unique_ptr<Expression>);
-std::unique_ptr<Expression> toFormatted(std::unique_ptr<Expression>);
+std::unique_ptr<Expression> assertion(type::Type, std::unique_ptr<Expression>,
+                                      optional<std::unique_ptr<Expression>> def = {});
```

Comment:
`std::unique_ptr` is already nullable, maybe just `std::unique_ptr<Expression> def = nullptr`